### ISSUE TITLE
`npm.bootstrap` fails when NPM is not installed

### ISSUE THREAD

**Issue by eliasp** [saltstack/salt#26882]
Using the state `npm.bootstrap` fails when NPM isn't installed yet:
```
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/site-packages/salt/minion.py", line 1161, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/site-packages/salt/modules/state.py", line 523, in highstate
        whitelist=kwargs.get('whitelist')
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2968, in call_highstate
        return self.state.call_high(high)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 2067, in call_high
        ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 1623, in call_chunks
        running = self.call_chunk(low, running, chunks)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 1858, in call_chunk
        running = self.call_chunk(chunk, running, chunks)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 1769, in call_chunk
        self._mod_init(low)
      File "/usr/lib/python2.7/site-packages/salt/state.py", line 612, in _mod_init
        self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
      File "/usr/lib/python2.7/site-packages/salt/utils/lazy.py", line 90, in __getitem__
        raise KeyError(key)
    KeyError: 'npm.bootstrap'
```

It doesn't help adding a `require` statement on the `npm` package to prevent this - this looks more like an issue in the module which blindly tries to run `npm` without properly handling its absence.

```
               Salt: 2015.5.3
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.4.6
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.3.1
               RAET: Not Installed
                ZMQ: 3.2.5
               Mako: Not Installed
            Tornado: Not Installed
```

**Comment by arroyoc** [comment]
@eliasp Thanks for your report.

**Comment by immunda** [comment]
Any news on this or how to work around it?

**Comment by rallytime** [comment]
I can reproduce this on the 2015.5 branch, but it is fixed on 2015.8.0.
```
# salt-call --local state.sls npm
[INFO    ] Loading fresh modules for state activity
[INFO    ] Fetching file from saltenv 'base', ** skipped ** latest already in cache u'salt://npm.sls'
[INFO    ] Running state [coffee-script] at time 16:28:03.249406
[ERROR   ] State 'npm.installed' was not found in SLS 'npm'
Reason: 'npm' __virtual__ returned False: 'npm' binary not found on system

local:
----------
          ID: coffee-script
    Function: npm.installed
      Result: False
     Comment: State 'npm.installed' was not found in SLS 'npm'
              Reason: 'npm' __virtual__ returned False: 'npm' binary not found on system
     Started:
    Duration:
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   0.000 ms
```
NPM isn't installed:
```
>>> from salt.utils import which
>>> print(which('npm'))
None
```
Versions report:
```
# salt --versions
Salt Version:
           Salt: 2015.8.0

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.5.0
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.6.4
      gitpython: Not Installed
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.9.0
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 4.1.5-x86_64-linode61
         system: Ubuntu 14.04 trusty
```

**Comment by rallytime** [comment]
I found the fix for this. This bug was fixed on 2015.8 with #25928. I have back-ported the fix to the 2015.5 branch just now with #27681.

We've already started the testing/packaging process for 2015.5.6, so this won't be fixed there, but will be fixed in the 2015.5 branch.
